### PR TITLE
Delete all items in trash bin

### PR DESCRIPTION
**Pull request by Aeredren**
fixes: https://github.com/nextcloud/calendar/issues/3171

Signed-off-by: Romain Lebrun Thauront <romain@framasoft.org>

### CODE REVIEW

File: src/components/AppNavigation/CalendarList/Trashbin.vue [hunk 1]
```diff
@@ -97,9 +97,14 @@
...
 						</table>
-						<p v-if="retentionDuration" class="footer">
-							{{ n('calendar', 'Elements in the trash bin are deleted after {numDays} day', 'Elements in the trash bin are deleted after {numDays} days', retentionDuration, { numDays: retentionDuration }) }}
-						</p>
+						<div class="footer">
+							<p v-if="retentionDuration">
+								{{ n('calendar', 'Elements in the trash bin are deleted after {numDays} day', 'Elements in the trash bin are deleted after {numDays} days', retentionDuration, { numDays: retentionDuration }) }}
+							</p>
+							<button @click="onEmptyTrashBin()">
+								{{ t('tasks','Empty trash bin') }}
```
Comment (raimund-schluessler) [inline src/components/AppNavigation/CalendarList/Trashbin.vue:105]:
```suggestion
								{{ t('calendar','Empty trash bin') }}
```

File: src/components/AppNavigation/CalendarList/Trashbin.vue [hunk 2]
```diff
@@ -257,6 +262,23 @@ export default {
 				showError(t('calendar', 'Could not restore calendar or event'))
 			}
 		},
+		onEmptyTrashBin() {
+			OC.dialogs.confirm(
+				t('tasks', 'Do you really want to empty the trash bin?'),
+				t('tasks', 'Empty trash bin'),
```
Comment (raimund-schluessler) [inline src/components/AppNavigation/CalendarList/Trashbin.vue:268]:
```suggestion
				t('calendar', 'Do you really want to empty the trash bin?'),
				t('calendar', 'Empty trash bin'),
```

**Comment by Aeredren** [comment]
I just ported the "Empty trash bin" button from my PR on Tasks.
I notice some difference in html/css in the implementation of the otherwise similar trash bin between task and calendar:

- Using table (calendar) instead of CSS grid layout (task)
- The modal is scrolable (calendar) instead of having only a scrolable table/grid css (task)

This last one result in having the Empty all button hidden when the trash bin is full (in calendar)

**Comment by raimund-schluessler** [comment]
> 
> 
> I just ported the "Empty trash bin" button from my PR on Tasks. I notice some difference in html/css in the implementation of the otherwise similar trash bin between task and calendar:
> 
>     * Using table (calendar) instead of CSS grid layout (task)
> 
>     * The modal is scrolable (calendar) instead of having only a scrolable table/grid css (task)
> 
> 
> This last one result in having the Empty all button hidden when the trash bin is full (in calendar)

Yes, I changed this in the trash bin implementation for Tasks so that only the items scroll, and didn't find the time to port it to the Calendar trash bin yet.

**Review by raimund-schluessler — APPROVED** [review]
Just squash your commits, please.

**Review by st3iny — APPROVED** [review]
Works! Thanks for your contribution.

**Pull request merged**